src/App.js

```diff
@@ -11,23 +18,82 @@ class App extends React.Component {
       cardDeck: makeShuffledDeck(),
       // currCards holds the cards from the current round
       currCards: [],
+      player1score: 0,
+      player2score: 0,
+      isCardDeckEmpty: false,
+      pastRounds: [],
     };
   }
 
+  gameRestarted = () => {
+    this.setState((state) => ({
+      isCardDeckEmpty: false,
+      cardDeck: makeShuffledDeck(),
+      currCards: [],
+      pastRounds: [
+        ...state.pastRounds,
+        [state.player1score, state.player2score],
+      ],
+      player1score: 0,
+      player2score: 0,
+    }));
+
+    console.log(this.state.pastRounds);
+  };
+
   dealCards = () => {
     this.setState((state) => ({
       // Remove last 2 cards from cardDeck
       cardDeck: state.cardDeck.slice(0, -2),
       // Deal last 2 cards to currCards
       currCards: state.cardDeck.slice(-2),
     }));
+
+    // Add score to the player that wins the round. No points for draw.
+    if (this.state.currCards[0].rank > this.state.currCards[1].rank) {
+      this.setState((state) => ({ player1score: state.player1score + 1 }));
+    } else if (this.state.currCards[0].rank < this.state.currCards[1].rank) {
+      this.setState((state) => ({ player2score: state.player2score + 1 }));
+    }
+
+    // Restart game when card deck is empty
+    if (this.state.cardDeck.length === 0) {
```

Comment:
when looking at arrays you can shorten the check by simply evaluating `if (!this.state.cardDeck.length)`

Reply:
so any non-zero number is true? and the ! operator will switch 0 which is false to true?